### PR TITLE
Write pointing unit tests

### PR DESCRIPTION
**Pull request by evanmayer**
I'm starting classes, so I won't have time to finish as many tests as I'd like to, but I want to get these in main so they don't go stale.

I'd say this is ~50% of the coverage I think is practical to get on `pointing.c`, but there's plenty to go on here in case someone else wants to pick up writing them, and this is also a good example of what I'd like to see more widespread in `mcp`.

As a bonus, it is an example of using `cmocka`'s `**state` pointer to create test fixtures and pass them to test functions, a feature which I found was not well documented.

These aren't expected to build in `main` yet. Please continue to compile with tests off (the default). This is because the `cmocka` build step requires a newer CMake than our Debian Jessie Vagrant VMs have. I was able to develop tests by manually installing a newer CMake from source, and all tests are passing. I expect these tests to bear fruit after we switch to Ubuntu 20.04.

### CODE REVIEW

File: mcp/pointing/tests/test_pointing.c [hunk 1]
```diff
@@ -2,24 +2,450 @@
...
+        .offset_gy = 1.0,
+        .FC = 1.0,
+        .n_solutions = 0,
+        .since_last = 0,
+        .fs = (struct FirStruct *) balloc(fatal, sizeof(struct FirStruct)),
+        .new_offset_ifel_gy = 0.0,
+        .int_ifel = 0.0,
+        .prev_sol_el = 45.0
+    };
+    // ElSol.fs =
```
Comment (evanmayer) [inline mcp/pointing/tests/test_pointing.c:38]: Remove commented code

File: mcp/pointing/tests/test_pointing.c [hunk 2]
```diff
@@ -2,24 +2,449 @@
...
+        .samp_weight = 1.0, // sample weight per sample
+        .sys_var = 0.5,  // sytematic variance - can't do better than this
+        .trim = 0.0, // externally set trim to solution
+        .last_input = 45.0, // last good data point
+        .ifroll_gy_int = 1.0, // integral of the gyro since the last solution
+        .ifyaw_gy_int = 1.0,// integral of the gyro since the last solution
+        .offset_ifroll_gy =  1.0, // offset associated with solution
+        .offset_ifyaw_gy = 1.0,
+        .FC = 1.0, // filter constant
+        .n_solutions = 0, // number of angle inputs
```
Comment (mxk62) [inline mcp/pointing/tests/test_pointing.c:71]: Just a nitpick, the spacing between the code and the inline comments is rather inconsistent. Please, fix.

Comment (evanmayer) [reply mcp/pointing/tests/test_pointing.c]: Fixes pushed

**Review by mxk62 — COMMENTED** [review]
Looks good to me.

**Pull request merged**